### PR TITLE
Adds `findconfig` and some more tests

### PR DESCRIPTION
**Pull request by peter1000**

### CODE REVIEW

File: src/Elements/utilities.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+["Helper functions."]
+
+"""
+Find the value for ``key`` associated with a node's ``n`` configuration.
```
Comment (MichaelHatherly) [inline src/Elements/utilities.jl]: > Find the value for ``key`` associated with a node's ``n`` configuration.

to

> Find the value for ``key`` associated with node ``n``'s configuration.

File: src/Elements/utilities.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Returns a ``Nullable{T}`` object. ``isnull`` must be called to determine whether
+an object was actually found or not.
+
+When the configuration of ``n`` does not contain the key ``key`` then all it's parents
+are searched in turn.
+"""
+function findconfig(n::Node, key::Symbol, T)
+    haskey(n.data, key) && return Nullable{T}(n.data[key])
+    # Stage 2: Parent's config.
```
Comment (MichaelHatherly) [inline src/Elements/utilities.jl]: Comment not really needed, especially if no "stage 1" is mentioned.

File: src/Elements/utilities.jl [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    while isdefined(n, :parent)
+        n = n.parent
+        haskey(n.data, key) && return Nullable{T}(n.data[key])
+    end
+    Nullable{T}()
+end
```
Comment (MichaelHatherly) [inline src/Elements/utilities.jl]: Needs an explicit call to ``convert`` for the ``Nullable`` constructors. ``Nullable{T}(convert(T, n.data[key]))``. [Here](https://github.com/MichaelHatherly/Docile.jl/blob/de29d20a5404f690cbf5b97586c246898a0baee9/src/Cache/utilities.jl#L24) for example.

File: src/Elements/utilities.jl (hunk 1 above)
Comment (peter1000) [inline src/Elements/utilities.jl]: done

File: src/Elements/utilities.jl (hunk 2 above)
Comment (peter1000) [inline src/Elements/utilities.jl]: Removed that part

**Comment by MichaelHatherly** [comment]
If we find a use for ``asnull`` outside of this particular area then yes, but lets leave it here for the moment.

(GitHub is showing me that the inline comments are outdated, but I can't seem to see the changes you've made.)

File: src/Elements/utilities.jl [hunk 4]
```diff
@@ -0,0 +1,18 @@
+["Helper functions."]
+
+"""
+Find the value for ``key`` associated with a node ``n``'s configuration.
+
+Returns a ``Nullable{T}`` object. ``isnull`` must be called to determine whether
+an object was actually found or not.
+"""
+function findconfig(n::Node, key::Symbol, T)
+    haskey(n.data, key) && return Nullable{T}(n.data[key])
```
Comment (MichaelHatherly) [inline src/Elements/utilities.jl]: Use ``asnull`` here as well.

**Comment by peter1000** [comment]
I'm out for today

**Comment by MichaelHatherly** [comment]
> I'm out for today

Thanks for the work. I'll squash this locally and commit it.

**Pull request closed without merging**